llvm-bazel/llvm-project-overlay/llvm/BUILD.bazel

```diff
@@ -68,7 +55,7 @@ llvm_target_asm_parsers = [
     "Sparc",
     "WebAssembly",
     "X86",
-]
+] if t in llvm_targets]
```

Comment:
It's a bit tricky...

So, if we do this, it'll let code build against the target, but using it will be a bit "exciting"....

Imagine you have a test `a_test` that depends on a library `a` which in turn depends on library `b`, and it is `b` that depends on NVPTX. The default initialize-all helper routines that `a_test` might use won't actually initialize the NVPTX target, and so `a_test` would have to add custom initialization to explicitly initialize NVPTX which seems ... surprising.

Weirdly, most of the time, things will "just work" because there are a bunch of redundant attempts to initialize targets, but there may be strange and mysterious failures here.

Anyways, I'm happy to go either way -- I mostly want to surface that this isn't a trivial pattern to follow. Let me know which you prefer.

Reply:
That sounds like a good reason to at least hold off until we actually have a use case. Fully disabling the targets (like not even configuring them) is the way the cmake build works, right? The issue with the current setup is going to be that MLIR has a hard dependency on NVPTX. I guess that's ok because it's also true today though. Ugh cuda...